inc/compat/themes/compat-theme-cartsy.php

```diff
@@ -0,0 +1,30 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Compatibility with theme: Cartsy (by Redq).
+ */
+class FluidCheckout_ThemeCompat_Cartsy extends FluidCheckout {
+
+	/**
+	 * __construct function.
+	 */
+	public function __construct() {
+		$this->hooks();
+	}
+
+
+
+	/**
+	 * Initialize hooks.
+	 */
+	public function hooks() {
+		if ( ! FluidCheckout_CheckoutPageTemplate::instance()->is_distraction_free_header_footer_checkout() ) {
+			// Prevent theme's page template from being replaced by FC checkout template
+			add_filter( 'fc_enable_checkout_page_template', '__return_false', 10 );
+		}
```

Comment:
- Medium: Use "bail" statements early in the code to avoid increased code complexity and multiple nested `if` statements.

With "bail" statements, it is easier to read the code and understand which conditions need to be met for that function to run.

If needed, create a new function where you can "bail" early and then call it from the `hooks()` function. See example below (not tested):

```php
/**
 * Initialize hooks.
 */
public function hooks() {
	// Checkout templates
	$this->checkout_layout_hooks();
}

/*
 * Checkout templates hooks.
 */
public function checkout_layout_hooks() {
	// Bail if using the distraction free template
	if ( FluidCheckout_CheckoutPageTemplate::instance()->is_distraction_free_header_footer_checkout() ) { return; }

	// Prevent theme's page template from being replaced by FC checkout template
	add_filter( 'fc_enable_checkout_page_template', '__return_false', 10 );
}
```